shell/platform/android/io/flutter/view/AccessibilityBridge.java

```diff
@@ -1581,15 +1581,30 @@ private void willRemoveSemanticsNode(SemanticsNode semanticsNodeToBeRemoved) {
     //                    for null'ing accessibilityFocusedSemanticsNode, inputFocusedSemanticsNode,
     //                    and hoveredObject.  Is this a hook method or a command?
     semanticsNodeToBeRemoved.parent = null;
+
+    if (semanticsNodeToBeRemoved.platformViewId != -1
+        && embeddedAccessibilityFocusedNodeId != null
+        && accessibilityViewEmbedder.platformViewOfNode(embeddedAccessibilityFocusedNodeId)
+            == platformViewsAccessibilityDelegate.getPlatformViewById(semanticsNodeToBeRemoved.platformViewId)) {
+      // If the currently focused a11y node is within a platform view that is
+      // getting removed: clear it's a11y focus.
+      sendAccessibilityEvent(
```

Comment:
I'm a little concerned about sending an event on behalf of the webview (e.g my concern is that it's internal state may think that it is still focused), do you know whether this is considered safe to do?

Reply:
Unfortunately, I don't have an authoritative answer to that question :(

In my experiments it all seems to work out. E.g. when you go back to the route with the webview by popping the route above it, focus behaves reasonable.

Reply:
I'll live with that if it's the best we got 😕 

The other approach I'd consider would be to see if we can make the webview send that event (e.g maybe by turning it invisible within the vd?)